packages/e2e-tests/specs/widgets/adding-widgets.test.js

```diff
@@ -328,21 +332,16 @@ async function getSerializedWidgetAreas() {
 async function cleanupWidgets() {
 	await visitAdminPage( 'widgets.php' );
 
-	await page.evaluate( () => {
-		const deleteButtons = document.querySelectorAll(
-			'#widgets-right .widget button.widget-control-remove'
-		);
-
-		deleteButtons.forEach( ( deleteButton ) => deleteButton.click() );
-	} );
+	let widget = await page.$( '.widgets-sortables .widget' );
 
-	await page.click( '#inactive-widgets-control-remove' );
+	// We have to do this one-by-one since there might be race condition when deleting multiple widgets at once.
+	while ( widget ) {
+		const deleteButton = await widget.$( 'button.widget-control-remove' );
+		const id = await widget.evaluate( ( node ) => node.id );
+		await deleteButton.evaluate( ( node ) => node.click() );
+		// Wait for the widget to be removed from DOM.
+		await page.waitForSelector( `#${ id }`, { hidden: true } );
 
-	await page.waitForFunction(
-		() =>
-			document.querySelectorAll( '.widgets-sortables .widget' ).length ===
-			0
-	);
-	// No idea why we need this, but just asserting the widgets are gone seems to be not enough.
-	await page.waitForTimeout( 500 );
+		widget = await page.$( '.widgets-sortables .widget' );
```

Comment:
While I think it's good to prefer UI-based actions in an e2e test, if this is getting complicated another option might be to dispatch an action removing all the blocks.